check_http_json.rb

```diff
@@ -18,23 +18,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-
 # Requires.
-require 'rubygems'
 require 'json'
 require 'net/http'
 require 'net/https'
 require 'uri'
 require 'optparse'
 require 'timeout'
 
+# Manage Nagios messages and exit code
+module Nagios
```

Comment:
I understand the interest of using a module here, and I've no qualm with the code you've produced; however, I'm curious as to the *why*.  As in: in this particular code base, why replace the otherwise functioning `do_exit` function with a module at all?

Part of the reason that everything is a function is because I wanted to keep the code simple and readable by sysadmins who might not otherwise be programmers. If one is coming from a world of, say, Bash, functions are easy to digest. By introducing a self-referencing class, as well as the getter & setter pattern, you're also introducing more advanced (and mildly Ruby-specific) concepts.

Again, this is something that I was actively trying to avoid in this particular code base. :smile: I'm open to debate on the topic though!